### PR TITLE
Better handling for nil receipts

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/adapters/eth_tx.go [hunk 1]
```diff
@@ -191,6 +191,11 @@ func addReceiptToResult(
 		}
 	}
 
+	if receipt == nil {
```
Comment (j16r) [inline core/adapters/eth_tx.go]: This should probably go at line 185 right? Save some work...

Comment (samsondav) [reply core/adapters/eth_tx.go]: @j16r Steve suggested this, but I kind of like having the nil check right next to the dereference.

**Comment by se3000** [comment]
@spooktheducks I feel like an integration test is a bit heavy handed here. At the very least. Maybe at least do it in another to get this merged in cleanly with https://github.com/smartcontractkit/chainlink/pull/2483?

File: core/adapters/eth_tx.go (hunk 1 above)
Comment (j16r) [reply core/adapters/eth_tx.go]: If you're adding a nil receipt to a result, this seems like a programming error, not a runtime error.

**Comment by samsondav** [comment]
Note that a similar fix was merged as part of 0.7.7.

Will need to resolve conflicts between these two.

**Comment by samsondav** [comment]
@spooktheducks Do you still feel strongly about an integration test on this?

**Pull request merged**